infrastructure/backup/README.md

```diff
@@ -0,0 +1,70 @@
+
+The main purpose of Velero in CrownLabs is to _mitigate_ the following eventualities:
+- failed/buggy updates
+- unintentional resources deletion
+- catastrophical physical cluster destruction/loss of all masters(*)
+a backup system can be of use.
+
+(*) this would require backups to be stored in a physically different/distant location relatively to the cluster.
+
+Notes:
+- saving backups on the cluster itself could help to mitigate unintentional deletions but it's not a great strategy, since deletion of volumes or related resources could lead in deletion of backups themselves.
+
+## How has been deployed
+
+The backup solution chosen for CrownLabs is based on [Velero](https://velero.io/). 
+
+Velero can store backups of Kubernetes resources and possibly associated volumes on several cloud native/proprietary storage solutions (such as AWS, GCP, Azure, VMWare, OpenStack, etc.).
+
+In case of full bare-metal environments, it's possible to use [Minio](https://min.io/), an open source, Kubernetes-native onbject storage solution which is compliant with Amazon's S3 API. This provides a valid storage target for Velero, provided that the mounted data path is not being stored on the cluster itself.
+
+The provided sample configuration includes the deployment of a Minio instance that uses an iSCSI target (hosted outside of the cluster) for storing data.
+
+Velero offers not only the possibility to backup resources but also the content of the volumes present on the cluster. This can be done rather automatically by mean of Volume Snapshotters but recent versions of Velero include an integration wirh [Restic](https://restic.net/) (that can be deployed automatically by Velero itself) which enables volumes snapshots without requiring snapshotters and regardless the storage class of the volumes to backup.
+
+### Deployment
+
+```bash
+helm repo add vmware-tanzu https://vmware-tanzu.github.io/helm-charts
+helm install velero vmware-tanzu/velero --namespace velero --create-namespace -f values.yaml --dry-run
+```
+
+[](https://github.com/vmware-tanzu/helm-charts/blob/main/charts/velero/README.md)
+
+## Usage
+
+It's better to use Velero's client in order to interact with its deployment on the cluster. On [the official documentation](https://velero.io/docs/) it's possible to see how to install the client, in particular [here](https://velero.io/docs/v1.6/basic-install/#install-the-cli) are the instructions for the currently installed version.
+
+### Prerequisites: volumes preparation
+
+The volume snapshotting of Velero through Restic can work on a whitelist or blacklist basis. In the case of CrownLabs it has been chosen to use the whitelist approach since crucial volumes to be persisted can be identified. In order to tell Velero to backup a volume it's necessary to add a special annotation to the pods to which such volumes are mounted: `backup.velero.io/backup-volumes=VOL_NAME_2_,VOL_NAME_2,...`. 
```

Comment:
```suggestion
The volume snapshotting of Velero through Restic can work on a whitelist or blacklist basis. In the case of CrownLabs it has been chosen to use the whitelist approach since crucial volumes to be persisted can be identified. In order to request Velero to backup a volume it is necessary to add a special annotation to the pods to which such volumes are mounted: `backup.velero.io/backup-volumes=VOL_NAME_2_,VOL_NAME_2,...`. 
```